### ISSUE TITLE
consider only writing dependencies in tags

### ISSUE THREAD

**Issue by dsschult** [WIPACrepo/wipac-dev-py-dependencies-action#4]
The dependency files create lots of spurious merge conflicts, so don't write them to `main`.  Instead, during the release process, write them to the local repo before a tag branch gets pushed.  This essentially adds files only to the tag.

Slack discussion: https://icecube-spno.slack.com/archives/CAH0NDF8B/p1713563069877149

The `assets` feature of python-semantic-release is supposed to do this:
https://python-semantic-release.readthedocs.io/en/latest/configuration.html#assets-list-str

**Comment by ric-evans** [comment]
NOTE: In the newest workflow edition (https://github.com/WIPACrepo/wipac-dev-actions-testbed/blob/main/.github/workflows/cicd.yml#L133), the action would be run preceding the github release step.

**TLDR:** No more dep files in the repo, put them in the GitHub release files (along with the wheels). This would lose git line history for the logs, which is like 50% of the reason for the action...

Thoughts:

Moving the action to that portion would mean longer-lived PRs (like https://github.com/icecube/skymap_scanner/pull/273) would not benefit from seeing updates in the dependencies.

1. This could be solved by printing the dependencies in a separate job (no file writing/re-writing) -- GHA logs only live ~30 days, but that's an okay tradeoff. Attention will need to be paid to any duplicated docker image build logic...

2. Alternatively, this action could use GHA artifacts (behind an `if: <branch is main>`). Then, the release action logic could attach the dep logs to the new release.

One thing is for sure: no one would complain about CI tests no longer being cancelled due to a bot auto-pushing an updated dep file 😄

**Comment by ric-evans** [comment]
New idea! Put the dep logs in a **separate branch called `py-dep-logs`** (for each repo) -- completely managed by this action. At the repo root, there are dep log files. **This maintains git line history with no merge conflicts ever.** The auto-commits would have the name of the version tag.

Nice to have: subdirs for ongoing feature branches that are deleted when the branch is deleted. Otherwise, the just-print option is okay.

**Comment by ric-evans** [comment]
I made a more elegant solution over in https://github.com/WIPACrepo/wipac-dev-py-dependencies-action/pull/10
